server/network.go

```diff
@@ -358,97 +339,127 @@ func (lc *localNetwork) loadSnapshot(
+// the Primary Network (P-Chain, X-Chain, C-Chain.)
+// Populates [lc.subnets] with all subnets that exist.
+// Doesn't contain the Primary network.
+// Assumes [lc.lock] is held.
 func (lc *localNetwork) updateSubnetInfo(ctx context.Context) error {
-	node, err := lc.nw.GetNode(lc.nodeNames[0])
+	nodeNames := maps.Keys(lc.nodeInfos)
+	node, err := lc.nw.GetNode(nodeNames[0])
 	if err != nil {
 		return err
 	}
+
 	blockchains, err := node.GetAPIClient().PChainAPI().GetBlockchains(ctx)
 	if err != nil {
 		return err
 	}
+
 	for _, blockchain := range blockchains {
-		if blockchain.Name != "C-Chain" && blockchain.Name != "X-Chain" {
-			lc.customChainIDToInfo[blockchain.ID] = chainInfo{
-				info: &rpcpb.CustomChainInfo{
-					ChainName: blockchain.Name,
-					VmId:      blockchain.VMID.String(),
-					SubnetId:  blockchain.SubnetID.String(),
-					ChainId:   blockchain.ID.String(),
-				},
-				subnetID:     blockchain.SubnetID,
-				blockchainID: blockchain.ID,
-			}
+		if blockchain.Name == "C-Chain" || blockchain.Name == "X-Chain" {
+			continue
+		}
+		lc.customChainIDToInfo[blockchain.ID] = chainInfo{
+			info: &rpcpb.CustomChainInfo{
+				ChainName: blockchain.Name,
+				VmId:      blockchain.VMID.String(),
+				SubnetId:  blockchain.SubnetID.String(),
+				ChainId:   blockchain.ID.String(),
+			},
+			subnetID:     blockchain.SubnetID,
+			blockchainID: blockchain.ID,
 		}
 	}
+
 	subnets, err := node.GetAPIClient().PChainAPI().GetSubnets(ctx, nil)
 	if err != nil {
 		return err
 	}
+
 	lc.subnets = []string{}
 	for _, subnet := range subnets {
 		if subnet.ID != avago_constants.PlatformChainID {
 			lc.subnets = append(lc.subnets, subnet.ID.String())
 		}
 	}
-	for _, nodeName := range lc.nodeNames {
-		nodeInfo := lc.nodeInfos[nodeName]
+
+	for _, nodeInfo := range lc.nodeInfos {
 		for chainID, chainInfo := range lc.customChainIDToInfo {
 			lc.log.Info(fmt.Sprintf(logging.LightBlue.Wrap("[blockchain RPC for %q] \"%s/ext/bc/%s\""), chainInfo.info.VmId, nodeInfo.GetUri(), chainID))
 		}
 	}
 	return nil
 }
 
-func (lc *localNetwork) waitForLocalClusterReady(ctx context.Context) error {
+// Assumes [lc.lock] isn't held.
+func (lc *localNetwork) AwaitHealthy(ctx context.Context) error {
+	lc.lock.Lock()
+	defer lc.lock.Unlock()
+
+	return lc.awaitHealthy(ctx)
+}
+
+// Returns nil when [lc.nw] reports healthy.
+// Updates node and subnet info.
+// Assumes [lc.lock] is held.
+func (lc *localNetwork) awaitHealthy(ctx context.Context) error {
+	if lc.nw == nil {
+		return ErrNetworkNotRunning
+	}
+
 	ux.Print(lc.log, logging.Blue.Wrap(logging.Bold.Wrap("waiting for all nodes to report healthy...")))
 
 	if err := lc.nw.Healthy(ctx); err != nil {
 		return err
 	}
 
-	for _, name := range lc.nodeNames {
-		nodeInfo := lc.nodeInfos[name]
+	if err := lc.updateNodeInfo(); err != nil {
+		return err
+	}
+
+	if err := lc.updateSubnetInfo(ctx); err != nil {
+		return err
+	}
+
+	// TODO do we need to log this? Seems verbose
```

Comment:
Open question for you guys

Reply:
Possibly at most Debug log

Reply:
I'll leave it to you guys to decide -- feel free to push a commit changing this as you see fit